src/device-registry/controllers/create-site.js

```diff
@@ -269,6 +269,47 @@ const manageSite = {
     }
   },
 
+  findNearestSiteByCoordinates: async (req, res) => {
+    try {
```

Comment:
Thanks @NoahNsimbe , glad we are handling errors here, awesome stuff!
Could you double check the relevance of the double "try catch statements" here?